test/jdk/java/time/test/java/time/TestInstant.java

```diff
@@ -151,4 +152,47 @@ public void test_until_1arg_NPE() {
             Instant.now().until(null);
         });
     }
+
+    @DataProvider
+    private Object[][] valid_instants() {
+        return new Object[][] {
+            {"2017-01-01T00:00:00.000+02"},
+            {"2017-01-01T00:00:00.000+0200"},
+            {"2017-01-01T00:00:00.000+02:00"},
+            {"2017-01-01T00:00:00.000+020000"},
+            {"2017-01-01T00:00:00.000+02:00:00"},
+            {"2017-01-01T00:00:00.000Z"},
+        };
+    }
+
+    @Test(dataProvider = "valid_instants")
+    public void test_parse_valid(String instant) {
+        Instant.parse(instant);
+    }
+
+    @DataProvider
+    private Object[][] invalid_instants() {
+        return new Object[][] {
+            {"2017-01-01T00:00:00.000"},
+            {"2017-01-01T00:00:00.000+0"},
+            {"2017-01-01T00:00:00.000+0:"},
+            {"2017-01-01T00:00:00.000+02:"},
```

Comment:
It is nice that leniency allows `0200` (i.e., missing white space), but not `02:` or `02:00:` (i.e., trailing white space).